library/coretests/tests/floats/mod.rs

```diff
@@ -20,6 +19,31 @@ impl Approx for f128 {
     const LIM: Self = 1e-9;
 }
 
+trait TestableFloat {
+    const SMALL_NORMAL: Self;
+    const SUBNORMAL: Self;
+}
+
+impl TestableFloat for f16 {
+    const SMALL_NORMAL: Self = 1e-4;
+    const SUBNORMAL: Self = 1e-5;
+}
+
+impl TestableFloat for f32 {
+    const SMALL_NORMAL: Self = 1e-37;
+    const SUBNORMAL: Self = 1e-38;
+}
+
+impl TestableFloat for f64 {
+    const SMALL_NORMAL: Self = 1e-307;
+    const SUBNORMAL: Self = 1e-308;
+}
+
+impl TestableFloat for f128 {
+    const SMALL_NORMAL: Self = 1e-4931;
+    const SUBNORMAL: Self = 1e-4932;
+}
```

Comment:
Could you rename `SMALL_NORMAL` to `MIN_POSITIVE_NORMAL`? Then these can just be `f32::MIN_POSITIVE` rather than a handwritten value. You could also use `MIN_POSITIVE` directly, but that name is confusing.

`SUBNORMAL` can similarly be renamed to `MAX_SUBNORMAL` with a value `f32::MIN_POSITIVE.next_down()`.